controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStream.java

```diff
@@ -114,6 +114,7 @@ class PravegaTablesStream extends PersistentStreamBase {
     // completed transactions key
     private static final String STREAM_KEY_PREFIX = "Key" + SEPARATOR + "%s" + SEPARATOR + "%s" + SEPARATOR; // scoped stream name
     private static final String COMPLETED_TRANSACTIONS_KEY_FORMAT = STREAM_KEY_PREFIX + "/%s";
+    private static final int MAX_RECORD_BY_LIST_TXN_API = 500;
```

Comment:
This perhaps could be a controller configuration property defaulted to 500 txns.

Reply:
Done as suggested.